### PR TITLE
Page to link users with students created

### PR DESCRIPTION
**Pull request by JParisFerrer**
Adds a page to link and view linked students for the logged in user. Also adds link to the navbar, and related auth check helper.

Security is virtually non-existent; to do this securely would need either a) shib and do this automatically, or b) some sort of complex email verification system

Also adds an `info` flash type with display in the header.

### CODE REVIEW

File: src/pages/course/course.js [hunk 1]
```diff
@@ -31,7 +31,11 @@ router.get(
     res.locals.courseDept = courseRow.dept
     res.locals.courseNumber = courseRow.number
     res.locals.courseName = courseRow.course_name
-    if (result.rows.length >= 1 && courseRow.name !== undefined && courseRow.name !== null) {
+    if (
+      result.rows.length >= 1 &&
+      courseRow.name !== undefined &&
+      courseRow.name !== null
```
Comment (Binary-Eater) [inline src/pages/course/course.js:37]: Thanks for making these explicit ;)

File: src/pages/link/link.ejs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        <div class="list-group">
+            <% if (linked_course_data.length > 0) {
+
+                linked_course_data.forEach( (cdata) => {  
+            %>
+        <div class="list-group-item d-flex justify-content-between align-items-center">
+           <%= cdata.stu_ci_name %> (<%= cdata.course_name %>, <%= cdata.stu_ci_term %> <%= cdata.stu_ci_year %>)
+        </div>
+        <%
```
Comment (Binary-Eater) [inline src/pages/link/link.ejs:20]: JSP++

File: src/pages/link/link.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+router.post(
+  '/',
+  asyncErrorHandler(async (req, res, _next) => {
+    if (!(await checks.isLoggedIn(req))) {
+      res.sendStatus(403)
+      return
+    }
+
+    const uin = Number.parseInt(req.body.uin, 10)
```
Comment (Binary-Eater) [inline src/pages/link/link.js:40]: **Warning:** `parseInt` acts like`strol` in C.

Example
```js
function roughScale(x, base) {
  var parsed = Number.parseInt(x, base);
  if (Number.isNaN(parsed)) {
    return 0;
  }
  return parsed ;
}

console.log(roughScale(' 0xF', 16));
// expected output: 15

console.log(roughScale('321 troll', 10));
// expected output: 321
```

Comment (Binary-Eater) [reply src/pages/link/link.js:40]: Actually, guessing you use `atoi` more but still same behavior.

Comment (Binary-Eater) [reply src/pages/link/link.js:40]: Hmmmm, you may actually want that behavior for handling the swipers if you actually care. Personally, I think you should restrict the string length and then parse as a universal solution (not sure what garbage the swiper puts after the UIN, like for example another number).

Comment (JParisFerrer) [reply src/pages/link/link.js:40]: not sure how much we care about this; We would also want/need to change all the other number inputs around the app to whatever method accepts if a real number, not just prefix

File: src/pages/link/link.ejs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        %>
+        </div>
+
+        <br/>
+        <h2> Relink Courses </h2>
+
+        <form method="POST">
+    	  <div class="form-group">
+    	    <label for="uin">UIN</label>
+    	    <input
```
Comment (Binary-Eater) [inline src/pages/link/link.ejs:36]: Should be an input of type `number`.

**Review by Binary-Eater — CHANGES_REQUESTED** [review]

File: src/pages/link/link.js (hunk 3 above)
Comment (Binary-Eater) [reply src/pages/link/link.js:40]: Ya, I agree. I didn't give this much thought before. Could make a PR to sweep through and check for this for inputs of type `number` passed to the `res.body`.

File: src/pages/link/link.ejs (hunk 4 above)
Comment (JParisFerrer) [reply src/pages/link/link.ejs:36]: done

**Review by Binary-Eater — APPROVED** [review]
Seems to be working.

**Pull request merged**